src/gui/processing/qgsprocessingmultipleselectiondialog.cpp

```diff
@@ -322,14 +326,27 @@ QgsProcessingMultipleInputPanelWidget::QgsProcessingMultipleInputPanelWidget( co
   connect( addDirButton, &QPushButton::clicked, this, &QgsProcessingMultipleInputPanelWidget::addDirectory );
   buttonBox()->addButton( addDirButton, QDialogButtonBox::ActionRole );
   setAcceptDrops( true );
+
+  // populate already selected layer values
```

Comment:
This is missing logic from setProject -- eg `if ( mParameter->layerType() != Qgis::ProcessingSourceType::File )`